### PR TITLE
Add a rake task to repopulate metrics

### PR DESCRIPTION
**Pull request by peteglondon**
Adds a rake task `rake etl:rerun_master`

This can be run across a range of dates and does the following:

* Delete the metrics for each day.
* Run the Etl::Master process for each day.

You can run the task as follows:

```
bundle exec rake etl:rerun_master[2018-10-01,2018-11-11]
```

Or if you are using zsh:

```
bundle exec rake etl:rerun_master\[2018-10-01,2018-11-11\]
```

Trello: https://trello.com/c/lDZPun1o/838-3-repopulate-data-from-1-oct-to-11-nov-2018#

### CODE REVIEW

**Comment by peteglondon** [comment]
[(3) Repopulate data from 1 Oct to 11 Nov 2018](https://trello.com/c/lDZPun1o/838-3-repopulate-data-from-1-oct-to-11-nov-2018)

File: lib/tasks/etl.rake [hunk 1]
```diff
@@ -15,6 +15,19 @@ namespace :etl do
     end
   end
 
+  desc 'Delete existing metrics and Run Etl Master process across a range of dates'
+  task :repopulate_master, %i[from to] => [:environment] do |_t, args|
```
Comment (pmanrubia) [inline lib/tasks/etl.rake]: Can we rename this task to `rerun_master` or something similar?

Comment (peteglondon) [reply lib/tasks/etl.rake]: Sure, give me a minute

File: lib/tasks/etl.rake [hunk 2]
```diff
@@ -15,6 +15,19 @@ namespace :etl do
     end
   end
 
+  desc 'Delete existing metrics and Run Etl Master process across a range of dates'
+  task :rerun_master, %i[from to] => [:environment] do |_t, args|
+    from = args[:from].to_date
+    to = args[:to].to_date
+    (from..to).each do |date|
+      puts "Deleting existing metrics for #{date}"
```
Comment (pmanrubia) [inline lib/tasks/etl.rake]: Can you think of a way to isolate each day in a transaction? This way if the process fails we don't leave inconsistencies

File: lib/tasks/etl.rake [hunk 3]
```diff
@@ -15,6 +15,19 @@ namespace :etl do
...
+      Facts::Metric.where(dimensions_date_id: date).delete_all
+      puts "Running Etl::Master process for #{date}"
+      Etl::Master::MasterProcessor.process(date: date)
+      puts "finished running Etl::Master for #{date}"
```
Comment (pmanrubia) [inline lib/tasks/etl.rake]: Are this `puts` also rendered when running tests?

**Comment by pmanrubia** [comment]
Looks good, two small comments

File: lib/tasks/etl.rake [hunk 4]
```diff
@@ -9,9 +9,24 @@ namespace :etl do
...
+      console_log "finished repopulating GA pviews for #{date}"
+    end
+  end
+
+  desc 'Delete existing metrics and Run Etl Master process across a range of dates'
+  task :rerun_master, %i[from to] => [:environment] do |_t, args|
+    from = args[:from].to_date
+    to = args[:to].to_date
+    (from..to).each do |date|
+      ActiveRecord::Base.transaction do
```
[the remark this answers is not in the document]
Comment (peteglondon) [reply lib/tasks/etl.rake]: Done!

**Pull request merged**